runtime/mount/agent.py

```diff
@@ -2981,6 +3042,23 @@ async def run_agent_loop(self) -> None:
 
             response_content = response.model_dump()["content"]
             if response_content is not None and (not isinstance(response_content, list) or len(response_content) > 0):
+                if isinstance(response_content, list):
+                    thinking_text = ""
+                    response_text = ""
+                    for block in response_content:
+                        if isinstance(block, dict):
+                            if block.get("type") == "thinking":
+                                thinking_text = block.get("thinking", "")
+                            elif block.get("type") == "text":
+                                response_text = block.get("text", "")
+
+                    if thinking_text:
+                        summary = await self._summarize_thinking(f"Thinking:\n{thinking_text}\n\nResponse:\n{response_text}", is_final=True)
```

Comment:
I would probably pass in the prompt rather than have a boolean is final. Just makes it a less loaded function and makes reading it here more clear.